### PR TITLE
[cuDNN] Add mhlo to cudnn stub pass.

### PR DESCRIPTION
**Pull request by chsigg**
A pass with a simple pattern to rewrite an stablehlo op to a cudnn op, plus a test.

### CODE REVIEW

File: compiler/src/openxla/compiler/nvgpu/test/example.mlir [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  %max = mhlo.constant dense<0xFFFFFFFF> : !tensor
+  // CHECK: cudnn.pointwise_relu(%0) type = f32 lower_clip = 0.000000e+00
+  // CHECK-SAME: -> !cudnn.tensor_desc<
+  // CHECK-SAME:      1x16x32x8xf32,
+  // CHECK-SAME:      alignment = 0,
+  // CHECK-SAME:      stride = [4096, 256, 8, 1]
+  // CHECK-SAME:    >
+  %result = mhlo.clamp %min, %argument, %max : !tensor
+  return %result : !tensor
+}
```
Comment (ezhulenev) [inline compiler/src/openxla/compiler/nvgpu/test/example.mlir:16]: newline

Comment (chsigg) [reply compiler/src/openxla/compiler/nvgpu/test/example.mlir]: Done.

File: compiler/src/openxla/compiler/nvgpu/Transforms/BUILD.bazel [hunk 2]
```diff
@@ -34,8 +34,10 @@ cc_library(
     deps = [
         ":PassesIncGen",
         "//compiler/src/openxla/compiler/nvgpu:defs",
+        "//compiler/src/openxla/compiler/nvgpu/Dialect/CUDNN/IR",
         "@llvm-project//mlir:IR",
         "@llvm-project//mlir:Pass",
+        "@mlir-hlo//:mlir_hlo",
```
Comment (jpienaar) [inline compiler/src/openxla/compiler/nvgpu/Transforms/BUILD.bazel:40]: Let's perhaps flip this to StableHLO for now (mlir-hlo repo is due for changes and I don't think the difference is meaningful for what we need here).

File: compiler/src/openxla/compiler/nvgpu/Transforms/ConvertMHLOToCUDNN.cpp [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      return rewriter.notifyMatchFailure(op, "expected constant min");
+    }
+    llvm::APFloat max;
+    if (!matchPattern(op.getMax().getDefiningOp(), m_ConstantFloat(&max)) ||
+        !max->isNan()) {
+      return rewriter.notifyMatchFailure(op, "expected NaN max");
+    }
+    TensorType tensor_type = op.getOperand().getType();
+    cudnn::TensorDescType tensor_desc_type = getTensorDescType(tensor_type);
+    Value input = rewriter.create<UnrealizedConversionCastOp>(
```
Comment (jpienaar) [inline compiler/src/openxla/compiler/nvgpu/Transforms/ConvertMHLOToCUDNN.cpp:64]: So here you are using the descriptor and the value as being castable, but the descriptor is the metadata of the lowered tensor (well, it is more the metadata of the memref really as these are in buffered world with explicit layouts and backing memory represented). Doesn't mean we can't change that, just considered this more of post bufferization level. But good to check here what folks are thinking.

File: compiler/src/openxla/compiler/nvgpu/test/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# Copyright 2023 The IREE Authors
```
Comment (jpienaar) [inline compiler/src/openxla/compiler/nvgpu/test/CMakeLists.txt:1]: I think we flipped authors in this repo.

**Review by jpienaar — COMMENTED** [review]
Overall looks good, only discussion point being about whether this is in bufferized world.

File: compiler/src/openxla/compiler/nvgpu/Transforms/BUILD.bazel (hunk 2 above)
Comment (chsigg) [reply compiler/src/openxla/compiler/nvgpu/Transforms/BUILD.bazel]: Done.

File: compiler/src/openxla/compiler/nvgpu/Transforms/ConvertMHLOToCUDNN.cpp (hunk 3 above)
Comment (chsigg) [reply compiler/src/openxla/compiler/nvgpu/Transforms/ConvertMHLOToCUDNN.cpp:64]: I didn't intend to imply these being castable. I wouldn't read anything into what the pass does, it's really just a placeholder.

File: compiler/src/openxla/compiler/nvgpu/test/CMakeLists.txt (hunk 4 above)
Comment (chsigg) [reply compiler/src/openxla/compiler/nvgpu/test/CMakeLists.txt]: Done.

**Pull request merged**